servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancer.java

```diff
@@ -331,64 +337,77 @@ public Completable closeAsyncGracefully() {
         return activeHosts.stream().map(Host::asEntry).collect(toList());
     }
 
-    private static class Host<Addr, C extends ListenableAsyncCloseable> implements AsyncCloseable {
+    private static final class Host<Addr, C extends ListenableAsyncCloseable> implements AsyncCloseable {
         @SuppressWarnings("rawtypes")
-        private static final AtomicReferenceFieldUpdater<Host, List> connectionsUpdater =
-                AtomicReferenceFieldUpdater.newUpdater(Host.class, List.class, "connections");
+        private static final AtomicReferenceFieldUpdater<Host, Object[]> connectionsUpdater =
+                AtomicReferenceFieldUpdater.newUpdater(Host.class, Object[].class, "connections");
 
         final Addr address;
-        private volatile List<C> connections = emptyList();
+        private volatile Object[] connections = EMPTY_ARRAY;
```

Comment:
How big is connections likely to be and how much contention is there likely to be? Could perhaps just use or extend CopyOnWriteArraySet

Reply:
Its internal `CopyOnWriteArrayList` uses `ReentrantLock` that we prefer to avoid as it may content on the event-loop threads.